src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1006,6 +1004,33 @@ static inline bool notification_match(reservation_t& res,
         ldpp_dout(res.dpp, 1) << "ERROR: failed to parse reservation id. error: " << ret << dendl;
         return ret;
       }
+    } else {
+      // no reservation - try to connect to endpoint instead
+      try {
+        const auto endpoint = RGWPubSubEndpoint::create(
+	  topic_cfg.dest.push_endpoint,
+	  topic_cfg.dest.arn_topic,
+	  RGWHTTPArgs(topic_cfg.dest.push_endpoint_args, dpp),
+	  dpp->get_cct());
+        ldpp_dout(res.dpp, 20) << "INFO: push endpoint created: "
+			       << topic_cfg.dest.push_endpoint << dendl;
+        const auto ret = endpoint->is_alive(dpp->get_cct(), res.yield);
+        if (ret < 0) {
+          ldpp_dout(dpp, 1)
+              << "ERROR: endpoint " << topic_cfg.dest.push_endpoint
+              << " is_alive failed. error: " << ret << dendl;
+          if (perfcounter) perfcounter->inc(l_rgw_pubsub_push_failed);
+          return ret;
+        }
+        if (perfcounter) perfcounter->inc(l_rgw_pubsub_push_ok);
```

Comment:
i don't think we should increment this one during publish_reserve(). publish_commit() increments it again at the end